### DISCUSSION TITLE
Manipulate only primary cursor in multicursor mode

### DISCUSSION THREAD

**Discussion by goyalyashpal** [helix-editor/helix#5949] [Q&A]
- Most of the time, the required mulicursor selection do have some common semantic/syntactic meaning, 
so can be done using same operations (moving  by words, to end of line, etc.)
- But often time, say one of thosee selections is the exception / odd one out and requires additional tweaking
- ability to move only primary cursor will help a lot in those cases...

It can be thought of as having a primary cursor mode in multicursor.

**Comment by the-mikedavis** [comment]
This is something that should be solved with marks (#3720). With the current bindings on that PR you could start with multiple selections, save those selections to a register with `save_selection_to_register` (`^s`), use `keep_primary_selection` (`,`) to remove all non-primary selections, operate on that primary selection, and then use `union_selection_from_register` (`^ca`) to restore the other selections.

**Reply by goyalyashpal** [reply]
it seems interesting, thanks for sharing 😃

**Comment by tahir-hassan** [comment]
This would be a good feature.  Perhaps you have a picker menu for selections and you have options to union those selections.
